unstoppable-ios-app/domains-manager-ios/Modules/QRScanner/ConnectedAppsList/ConnectedAppsListViewPresenter.swift

```diff
@@ -84,20 +84,21 @@ private extension ConnectedAppsListViewPresenter {
                 let apps = apps.sorted(by: { $0.appName < $1.appName })
                 guard let displayInfo = walletsDisplayInfo.first(where: { $0.address == apps[0].walletAddress }) else { continue }
                 
-                let items: [ConnectedAppsListViewController.Item] = apps.compactMap({ app in
-                    guard let domain = domains.first(where: { $0.name == app.domain.name }) else {
-                        Task {
-                            try await walletConnectServiceV2.disconnect(app: app)
-                        }
-                        return nil
+                let items: [ConnectedAppsListViewController.Item] = apps.map({ app in
+                    let domain: DomainItem
+                    if let _domain = domains.first(where: { $0.name == app.domain.name }) {
+                        domain = _domain
+                    } else {
+                        Debugger.printFailure("Forced to display a domain that has been disconnected, \(app.domain.name)", critical: true)
+                        domain = DomainItem(name: "Disconnected: \(app.domain.name)")
                     }
```

Comment:
We can now take domain directly from connected app?

Reply:
Yes, we have a property `domain` as `DomainItem`